src/test-support/client.ts

```diff
@@ -0,0 +1,78 @@
+type HistoryEntry = {
+  url: URL
+  request: {
+    method: string
+    body: Record<string, unknown>
+  }
+}
+type Request = {
+  url: string,
+  method: string,
+  searchParams: Record<string, string>,
+  body: Record<string, unknown>
+}
+type Listener = {
+  request: Request
+  resolve: (bool: boolean) => void
+}
```

Comment:
Some of these types are a bit awkward and I might be able to combine two of them, I'll look at that in a follow up.